pkg/shell/cockpit-docker.js

```diff
@@ -773,29 +774,251 @@ PageRunImage.prototype = {
         this.containers = null;
     },
 
+    /*
+     * validation functionality for the run image dialog
+     *
+     * warning:
+     *   - an expose port row has only one entry (container or host), the other will be flagged
+     *   - a linked container has no alias or an alias is given for no link
+     *
+     * error:
+     *   - a port is used more than once (same port/protocol exposed on container, same port/protocol used on host)
+     *   - a port number is invalid
+     *   - an alias for a linked container is used more than once
+     *
+     * any errors will result in a disabled 'run' button
+     */
+    update: function(behavior, section) {
+        function check_entries_valid() {
+            /* disable run button if there are any errors on the page */
+            $('#containers-run-image-run').prop('disabled', $('#containers_run_image_dialog').find('.has-error').length > 0);
+        }
+
+        function help_item_for_control(control, help_index) {
+            if (help_index === undefined)
+                return $(control.parent().parent().find('.help-block'));
```

Comment:
> Would .siblings() make this seem any less insane?
> http://api.jquery.com/siblings/
Siblings only takes exactly those into account, but the help-blocks are actually further down in the hierarchy - thus find(). I'll check the exact relation and see if we can simplify. The operations themselves aren't bad, since we're looking at a rather small set of objects each call.